### PR TITLE
Integrate Milton link processor

### PR DESCRIPTION
**Pull request by MattX**
This makes Coffebot check every message it sees for links and send said links to Milton.

I've had to turn the `Ignored` message type into a `Passive` type, and add a new type of handler for these messages. Currently, passive handlers won't look at `Valid` messages. This is fine for Milton, but I don't know if it's what we want in general.

Another possibly problematic point is that Milton requests can take a while, between cold-start times, page conversion, and sending the page chunks to the search engine, so maybe we want to revisit sequential processing in the future. I don't think this is a serious concern in the near term.

### CODE REVIEW

File: src/main/kotlin/coffeebot/commands/Milton.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.net.URL
+
+class MiltonClient {
+    private val httpClient = HttpClient(OkHttp) {
+        install(JsonFeature) {
+            serializer = GsonSerializer()
+        }
+        install(HttpTimeout) {
+            requestTimeoutMillis = 45_000
+            socketTimeoutMillis = 45_000
```
Comment (gabesaba) [inline src/main/kotlin/coffeebot/commands/Milton.kt:25]: 45 secs seems a bit high

File: src/main/kotlin/coffeebot/commands/Milton.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+private val miltonClient = MiltonClient()
+private val urlRegex = "https?://[\\w\\d:#@%/;$()~_?+-=\\\\.&]*".toRegex(RegexOption.IGNORE_CASE)
+
+val milton = PassiveCommand { message ->
+    val matches = urlRegex.findAll(message.contents).toList()
+    val results = matches.map { match ->
+        val urlValue = match.value
+        println(urlValue)
```
Comment (gabesaba) [inline src/main/kotlin/coffeebot/commands/Milton.kt:46]: I'm fine with leaving print statements as log statements. This is intent, right? (not relic of debugging)

File: coffeebot [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # /bin/bash
-./gradlew --build-cache installDist
+./gradlew --build-cache installDist &&
```
Comment (gabesaba) [inline coffeebot:2]: does this still run or do we need to indicate continued line ```\``` ?

File: src/main/kotlin/coffeebot/commands/Milton.kt (hunk 1 above)
Comment (gabesaba) [reply src/main/kotlin/coffeebot/commands/Milton.kt:25]: Unless MIlton's cold start takes that long?

Comment (MattX) [reply src/main/kotlin/coffeebot/commands/Milton.kt:25]: Cold starts do take very long, but I think 25 seconds would be enough. I'll change that.

File: src/main/kotlin/coffeebot/commands/Milton.kt (hunk 2 above)
Comment (MattX) [reply src/main/kotlin/coffeebot/commands/Milton.kt:46]: No it's clearly the rusty remains of printf-based debugging :p

File: coffeebot (hunk 3 above)
Comment (MattX) [reply coffeebot:2]: No, && acts as a line continuer in bash!

**Pull request merged**